stdlib/REPL/src/REPL.jl

```diff
@@ -194,6 +194,7 @@ function modules_to_be_loaded(ast::Expr, mods::Vector{Symbol} = Symbol[])
         end
     end
     for arg in ast.args
+        arg == Symbol("@eval") && break # don't search beyond an `@eval`
```

Comment:
The same kind of situation will also happen for `Core.eval(xxx, ...)` as well as `xxx.eval(...)`.

Reply:
@simeonschaub's suggestion below seems clean so I've gone for that